asciidoctorj-core/src/main/java/org/asciidoctor/jruby/ast/impl/FootnoteImpl.java

```diff
@@ -0,0 +1,45 @@
+package org.asciidoctor.jruby.ast.impl;
+
+import org.jruby.RubyStruct;
+
+import org.asciidoctor.ast.Footnote;
+import org.jruby.javasupport.JavaEmbedUtils;
+
+public class FootnoteImpl implements Footnote {
```

Comment:
Given the interface only allows getters, we could make this immutable with "final" attributes and a public "all args" constructor. That's assuming we don't want to expose setters, which I honestly don't know if we should or should not.

Reply:
Looking into more detail I see this deviates from other implementations because we are not keeping the reference with the rubyObject, nor we are using `RubyObjectWrapper` as base class to hande conversion.
The implementation works, but won't be able to add setters in the future and debugging may be a bit harder.